### PR TITLE
Add new tool `lobster-cpptest`

### PR DESCRIPTION
**Pull request by phiwuu**
Issue: #57

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,9 @@ meta_dist
 
 test_install
 test_install_monolithic
+lobster/tools/coda/_bazel_*
```
Comment (phiwuu) [inline .gitignore:12]: please undo these changes

File: create_adp_online_report.bat [hunk 2]
```diff
@@ -0,0 +1,4 @@
+rem This script runs LOBSTER on windows. its genrates the HTML report.
```
Comment (phiwuu) [inline create_adp_online_report.bat:1]: please delete this file

File: lobster/tools/coda/coda.py [hunk 3]
```diff
@@ -0,0 +1,355 @@
+#!/usr/bin/env python3
+#
+# lobster_coda - Extract C/C++ tracing tags from commands for LOBSTER
+# Copyright (C) 2023 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
```
Comment (phiwuu) [inline lobster/tools/coda/coda.py:4]: please use 2024 as copyright year

File: lobster/tools/coda/parser/constants.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+VALID_TESTMETHODS = [
+    "TM_EQUIVALENCE",
+    "TM_PAIRWISE",
+    "TM_GUESSING",
+    "TM_BOUNDARY",
+    "TM_CONDITION",
+    "TM_REQUIREMENT",
+    "TM_TABLE",
+    "TM_BOUNDARY",
```
Comment (phiwuu) [inline lobster/tools/coda/parser/constants.py:32]: `TM_BOUNDARY` exists twice

File: lobster/tools/coda/parser/constants.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+DEFECT = re.compile(
+    r"(@defect\s+)(((?:(CB-#\d+)|(OCT-#\d+)),?\s*)+)" + r"(?:///|/)\s+(((?:(CB-#\d+)|(OCT-#\d+)),?\s)+)?"
+)
+BRIEF = re.compile(r"(@brief\s+)([^@]+)")
+VERSION = re.compile(r"(@version\s+)(\d+([,]? \d+)*)+")
+OCT_TAG = r"(OCT-#\d+)"
+TESTMETHODS = re.compile(r"(@testmethods\s+)([^@]+)")
+# unmatch whole testmethod if invalid method is used
+# TESTMETHODS = re.compile(r"(@testmethods\s+)((" + "|".join(VALID_TESTMETHODS) + ")([,]? (" + "|".join(VALID_TESTMETHODS) + "))*)+")
+TEST = re.compile(r"(@test\s+)([^@]+)")
```
Comment (phiwuu) [inline lobster/tools/coda/parser/constants.py:58]: python files should end with `\n` (empty line)

File: lobster_adp_report.config
Comment (TannazVhdBMWExt) [inline lobster_adp_report.config:1]: should the file be removed?

File: lobster/tools/coda/coda.py [hunk 6]
```diff
@@ -0,0 +1,391 @@
...
+    dict
+        The dictionary containing the configuration for coda.
+
+    Raises
+    ------
+    Exception
+        If the config dict does not contain the required keys
+        or contains not supported values.
+    """
+    if not os.path.isfile(file_name):
```
[the remark this answers is not in the document]
Comment (phiwuu) [reply lobster/tools/coda/coda.py:90]: Refactoring is considered as "nice to have".

File: lobster/tools/coda/parser/constants.py [hunk 7]
```diff
@@ -0,0 +1,67 @@
+import re
+
+# URLs
+GIT_BASEDIR = "/vispiron-swe"
+GITHUB_URL = "https://github.com/"
+CODEBEMAER_URL = "https://codebeamer.bmwgroup.net/cb"
```
Comment (phiwuu) [inline lobster/tools/coda/parser/constants.py:6]: - typo in name
- please make this value configurable

File: lobster/tools/coda/coda.py [hunk 8]
```diff
@@ -0,0 +1,391 @@
...
+]
+
+map_test_type_to_key_name = {
+    RequirementTypes.REQS.value: 'requirements',
+    RequirementTypes.REQ_BY.value: 'required_by',
+    RequirementTypes.DEFECT.value: 'defect_tracking_ids',
+}
+
+
+def parse_config_file(file_name: str) -> dict:
```
Comment (phiwuu) [inline lobster/tools/coda/coda.py:64]: please add another feature to read the codebeamer ULR, in order to replace the `CODEBEAMER_URL` constant in `constants.py`

**Review by Sebastian-Koerner — APPROVED** [review]

**Review by Sebastian-Koerner — APPROVED** [review]

**Review by Sebastian-Koerner — APPROVED** [review]
